src/Worker.Extensions.DurableTask/DurableTaskClientExtensions.cs

```diff
@@ -18,6 +20,76 @@ namespace Microsoft.Azure.Functions.Worker;
 /// </summary>
 public static class DurableTaskClientExtensions
 {
+    /// <summary>
+    /// 
+    /// </summary>
+    /// <param name="client">The <see cref="DurableTaskClient"/>.</param>
+    /// <param name="request">The HTTP request that this response is for.</param>
+    /// <param name="instanceId">The ID of the orchestration instance to check.</param>
+    /// <param name="cancellation">The cancellation token.</param>
+    /// <param name="timeout">Total allowed timeout for output from the durable function. The default value is 10 seconds.</param>
+    /// <param name="retryInterval">The timeout between checks for output from the durable function. The default value is 1 second.</param>
+    /// <param name="returnInternalServerErrorOnFailure">Optional parameter that configures the http response code returned. Defaults to <c>false</c>.</param>
+    /// <returns></returns>
+    public static async Task<HttpResponseData> WaitForCompletionOrCreateCheckStatusResponseAsync(this DurableTaskClient client,
+        HttpRequestData request,
+        string instanceId,
+        CancellationToken cancellation = default,
+        TimeSpan? timeout = null,
```

Comment:
Also, per our discussion, let's use a `CancellationToken` parameter instead of using a timeout. Per .NET convention, we'll want to make this the last parameter in this method.